### PR TITLE
refactor(NewsChannel)!: rename NewsChannel to AnnouncementChannel

### PR DESCRIPTION
**Pull request by imnaiyar**
**Please describe the changes this PR makes and why it should be merged:**
Renames `NewsChannel` to `AnnouncementChannel` to be more in line with API naming (As it was renamed).

Related to:-
- https://github.com/discord/discord-api-docs/pull/5322
- https://github.com/discord/discord-api-docs/pull/5330

I imagine this would be a breaking change

BREAKING CHANGE: Rename `NewsChannel` to `AnnouncementChannel`

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

File: packages/discord.js/src/structures/Guild.js [hunk 1]
```diff
@@ -697,15 +697,16 @@ class Guild extends AnonymousGuild {
...
+   * @property {?(TextChannel|AnnouncementChannel|VoiceChannel|StageChannel|ForumChannel|MediaChannel)} channel
    * The widget invite channel
    */
 
   /**
    * The Guild Widget Settings object
    * @typedef {Object} GuildWidgetSettingsData
    * @property {boolean} enabled Whether the widget is enabled
-   * @property {?(TextChannel|NewsChannel|VoiceChannel|StageChannel|ForumChannel|MediaChannel|Snowflake)} channel
+   * @property {?(TextChannel|AnnouncementChannel|VoiceChannel|StageChannel|ForumChannel|
```
Comment (vladfrangu) [inline packages/discord.js/src/structures/Guild.js:708]: Double check the docs don't break with this going multiline

Comment (imnaiyar) [reply packages/discord.js/src/structures/Guild.js:708]: Looks good 
Screenshot 2024-10-06 224951

Perhaps I should disable the rule for this one, idk if it looks okay in the JSDoc itself

Comment (almeidx) [reply packages/discord.js/src/structures/Guild.js:708]: This is already done in plenty of places.

File: packages/discord.js/src/structures/AnnouncementChannel.js [hunk 2]
```diff
@@ -5,10 +5,10 @@ const BaseGuildTextChannel = require('./BaseGuildTextChannel');
...
 /**
- * Represents a guild news channel on Discord.
+ * Represents a guild announcement channel on Discord.
  * @extends {BaseGuildTextChannel}
  */
-class NewsChannel extends BaseGuildTextChannel {
+class AnnouncementChannel extends BaseGuildTextChannel {
   /**
    * Adds the target to this channel's followers.
    * @param {TextChannelResolvable} channel The channel where the webhook should be created
```
Comment (almeidx) [inline packages/discord.js/src/structures/AnnouncementChannel.js:14]: Missed the `@returns` tag in this JSDoc

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by Jiralite — APPROVED** [review]
Approved.

**Review by almeidx — APPROVED** [review]

**Review by didinele — APPROVED** [review]
assuming nothing was missed, lgtm

File: packages/discord.js/src/structures/Guild.js (hunk 1 above)
Comment (didinele) [reply packages/discord.js/src/structures/Guild.js:698]: resolving, this is ok IMO.

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**